Library/Homebrew/cmd/link.rb

```diff
@@ -40,18 +42,30 @@ def link
       verbose:   args.verbose?,
     }
 
-    args.named.to_kegs.each do |keg|
+    kegs = if args.HEAD?
+      args.named.to_kegs.group_by(&:name).map do |name, resolved_kegs|
+        head_keg = resolved_kegs.find { |keg| keg.version.head? }
+        if head_keg.blank?
+          opoo <<~EOS
+            No HEAD keg installed for #{name}
+            To install, run:
+              brew install --HEAD #{name}
+          EOS
+        end
+        head_keg
```

Comment:
```suggestion
        next head_keg if head_keg.present?

        opoo <<~EOS
          No HEAD keg installed for #{name}
          To install, run:
            brew install --HEAD #{name}
        EOS
```

Another option if you like this better. I'm happy either way

Reply:
I like this. Thanks for the edit. :)